### DISCUSSION TITLE
Access `setCenter()` when using Zustand

### DISCUSSION THREAD

**Discussion by wlabarron** [xyflow/xyflow#4031] [Q&A] [answered]
I've set up React Flow with Zustand using the instructions in [the documentation](https://reactflow.dev/learn/advanced-use/state-management). How can I now access the [`ReactFlowInstance`](https://reactflow.dev/api-reference/types/react-flow-instance) to manipulate the viewport with the [`setCenter`](https://reactflow.dev/api-reference/types/react-flow-instance#set-center) function? I've tried using [`useReactFlow()`](https://reactflow.dev/api-reference/hooks/use-react-flow), but get the error "Error: [React Flow]: Seems like you have not used zustand provider as an ancestor. Help: https://reactflow.dev/error#001", and I'm not sure where to go next.

**Comment by moklick** [comment]
you could add the instance or just the function to the store.

**Reply by wlabarron** [reply]
Within `store.ts`, how would you access the instance or function?

**Reply by moklick** [reply, marked as answer]
you would need to add it first. This is a possible solution: https://codesandbox.io/p/sandbox/xenodochial-violet-8m63x8?file=/App.tsx:32,43

**Reply by wlabarron** [reply]
Sorry, I'm getting "Sandbox not found" on that link. Could you share again please?

**Reply by moklick** [reply]
please try again. I am still not used to the recent csb changes..

**Reply by wlabarron** [reply]
That's working now, thank you! I'd missed using the `onInit` function on the `ReactFlow` element to initially get the instance. Thank you for your help!
